api/core/v1beta2/machine_types.go

```diff
@@ -87,6 +87,16 @@ const (
 
 	// ManagedNodeAnnotationDomain is one of the CAPI managed Node annotation domains.
 	ManagedNodeAnnotationDomain = "node.cluster.x-k8s.io"
+
+	// MachinePendingAcknowledgeMoveAnnotationName is an internal annotation added by the MS controller to a machine when being
+	// moved from the oldMS to the newMS. The annotation is removed as soon as the MS controller get the acknowledgment about the
+	// replica being accounted from the corresponding MD.
+	// Note: the annotation is added when reconciling the oldMS, and it is removed when reconciling the newMS.
+	MachinePendingAcknowledgeMoveAnnotationName = "in-place-updates.internal.cluster.x-k8s.io/pending-acknowledge-move"
```

Comment:
I think the constant names should match up with the string value (e.g. there's no machine in the annotation name). We usually also don't use a AnnotationName suffix, just Annotation

(same for other annotations added in this PR)

(maybe let's just finalize this together)